File: src/test/java/test/java/DatPerformanceTest.java
package test.java;

import me.saro.dat.DatUtils;
import me.saro.dat.crypto.DatCryptoAlgorithm;
import me.saro.dat.dat.DatCertificate;
import me.saro.dat.dat.DatManager;
import me.saro.dat.signature.DatSignatureAlgorithm;
import org.junit.jupiter.api.Test;

public class DatPerformanceTest {
    public DatCertificate generate(DatSignatureAlgorithm signatureAlgorithm, DatCryptoAlgorithm cryptoAlgorithm) {
        return DatCertificate.generate(
                0,
                signatureAlgorithm,
                cryptoAlgorithm,
                System.currentTimeMillis() - 10,
                System.currentTimeMillis() + 600,
                60
        );
    }

    @Test
    public void test() {
        String plain = DatUtils.generateRandomBase62(100);
        String secure = DatUtils.generateRandomBase62(100);

        System.out.println("plain : " + plain);
        System.out.println("secure : " + secure);

        for (var signatureAlgorithm : DatSignatureAlgorithm.getEntries()) {
            for (var cryptoAlgorithm : DatCryptoAlgorithm.getEntries()) {
                DatCertificate cert = generate(signatureAlgorithm, cryptoAlgorithm);
                String tag = signatureAlgorithm.name() + "/" + cryptoAlgorithm.name();
                String dat = "";

                long time = System.currentTimeMillis();
                for (var i = 0; i < 10000; i++) {
                    dat = DatManager.issue(cert, plain, secure);
                }
                System.out.println(tag + " issue * 10000 : " + (System.currentTimeMillis() - time) + " ms");

                time = System.currentTimeMillis();
                for (var i = 0; i < 10000; i++) {
                    DatManager.parse(cert, dat);
                }
                System.out.println(tag + " parse * 10000 : " + (System.currentTimeMillis() - time) + " ms");
            }
        }

    }
}

File: src/test/kotlin/test/kt/DatPerformanceTest.kt
package test.kt

import me.saro.dat.DatUtils.Companion.generateRandomBase62
import me.saro.dat.crypto.DatCryptoAlgorithm
import me.saro.dat.dat.DatCertificate
import me.saro.dat.dat.DatCertificate.Companion.generate
import me.saro.dat.dat.DatManager.Companion.issue
import me.saro.dat.dat.DatManager.Companion.parse
import me.saro.dat.signature.DatSignatureAlgorithm
import org.junit.jupiter.api.Test

class DatPerformanceTest {
    fun generate(signatureAlgorithm: DatSignatureAlgorithm, cryptoAlgorithm: DatCryptoAlgorithm): DatCertificate {
        return generate(
            0,
            signatureAlgorithm,
            cryptoAlgorithm,
            System.currentTimeMillis() - 10,
            System.currentTimeMillis() + 600,
            60
        )
    }

    @Test
    fun test() {
        val plain = generateRandomBase62(100)
        val secure = generateRandomBase62(100)

        println("plain : " + plain)
        println("secure : " + secure)

        for (signatureAlgorithm in DatSignatureAlgorithm.entries) {
            for (cryptoAlgorithm in DatCryptoAlgorithm.entries) {
                val cert = generate(signatureAlgorithm, cryptoAlgorithm)
                val tag: String = signatureAlgorithm.name + "/" + cryptoAlgorithm.name
                var dat = ""

                var time = System.currentTimeMillis()
                for (i in 0..9999) {
                    dat = issue(cert, plain, secure)
                }
                println(tag + " issue * 10000 : " + (System.currentTimeMillis() - time) + " ms")

                time = System.currentTimeMillis()
                for (i in 0..9999) {
                    parse(cert, dat)
                }
                println(tag + " parse * 10000 : " + (System.currentTimeMillis() - time) + " ms")
            }
        }
    }
}
